python/dataproc_templates/gcs/text_to_bigquery.py

```diff
@@ -119,19 +122,26 @@ def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
         input_file_codec_format: str = args[constants.TEXT_INPUT_COMPRESSION]
 
         logger.info(
-            "Starting GCS to Bigquery spark job with parameters:\n"
+            "Starting GCS to Bigquery Spark job with parameters:\n"
             f"{pprint.pformat(args)}"
         )
 
         # Read
-        input_data: DataFrame
-
-        input_data = spark.read\
-            .option(constants.HEADER, True) \
-            .option(constants.INFER_SCHEMA, True) \
-            .option(constants.INPUT_COMPRESSION, input_file_codec_format)\
-            .option(constants.INPUT_DELIMITER, input_delimiter)\
-            .csv(input_file_location)
+        spark_option_overrides = {}
+        if input_delimiter:
```

Comment:
Can't we just remove then the TEXT_INPUT_DELIMITER and TEXT_INPUT_COMPRESSION?
And keep using CSV_SEP and INPUT_COMPRESSION here?
Because this "spark_option_overrides" seems to alter the generic method to fit one exception, and the complexity is increasing.

Reply:
That was my first thought. But Shashank seemed keen that this is not a breaking change. Removing those existing options would break any existing scripts that use the template. I'd be happy to remove them though. It seems like the compression option for reading is non-existant anyway.

Reply:
Since we are adding spark options, which enables GCS related templates to accept CSV options like compression type and delimiter, I don't see a reason for the template "Text to BigQuery" to exist at all anymore.

The "GCS to BigQuery" would achieve the same results, and we just developed the "Text" template in the first days because Dataflow templates had it, but we made the "GCS to XXX / XXX to GCS" templates to accept different types of files like avro, parquet, csv and text itself, specially now accepting the spark options.

Additionally, it is not a template heavily used, so I think we should deprecate it.
Finally, deprecating it also would remove the need for this override.

What are your thoughts on this? @nj1973, @shashank-google 

Thanks!
Cheers

Reply:
Yes, I agree that this template is a duplicate of GCS to BigQuery and I agree it could be removed.

This would also impact these issues:

- https://github.com/GoogleCloudPlatform/dataproc-templates/issues/708
- https://github.com/GoogleCloudPlatform/dataproc-templates/issues/519

Do we deprecate it as part of this work or raise a new issue for it?

Reply:
@NiloFreitas May be a separate issue dedicatedly mentioning about the deprecating "Text to BigQuery" can be a good way.

Reply:
@NiloFreitas I've removed the extra complexity caused by supporting the text to BigQuery template and tests have all passed. Hopefully ready for another review.